daemon/daemon_unix.go

```diff
@@ -608,13 +609,13 @@ func configureKernelSecuritySupport(config *Config, driverName string) error {
 	return nil
 }
 
-func (daemon *Daemon) initNetworkController(config *Config, activeSandboxes map[string]interface{}) (libnetwork.NetworkController, error) {
+func (daemon *Daemon) initNetworkController(config *Config, plugingetter getter.PluginGetter, activeSandboxes map[string]interface{}) (libnetwork.NetworkController, error) {
 	netOptions, err := daemon.networkOptions(config, activeSandboxes)
 	if err != nil {
 		return nil, err
 	}
 
-	controller, err := libnetwork.New(netOptions...)
+	controller, err := libnetwork.New(plugingetter, netOptions...)
```

Comment:
You dont have to change the libnetwork.New API either. Pls refer to https://github.com/anusha-ragunathan/docker/blob/75cb7eb3f858b5bc04f1cb6fe0892cc35bce84ec/daemon/daemon.go#L1115 and just use the `networkOptions` (as you can see above) and populate the Options so that Libnetwork can pick that up.